### ISSUE TITLE
[Bug]: Desktop capture for Retina (HiDPI) screens uses 1x resolution

### ISSUE THREAD

**Issue by rf-figma** [electron/electron#44187]
### Preflight Checklist

- [x] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/main/CONTRIBUTING.md) for this project.
- [x] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.
- [x] I have searched the [issue tracker](https://www.github.com/electron/electron/issues) for a bug report that matches the one I want to file, without success.

### Electron Version

32.2.0

### What operating system(s) are you using?

macOS

### Operating System Version

Sonoma 14.5

### What arch are you using?

arm64 (including Apple Silicon)

### Expected Behavior

When using `getUserMedia` or `getDisplayMedia` to capture a Retina screen on macOS, the image or video should have the screen's full physical resolution.

### Actual Behavior

In practice, I'm getting image/video data with the screen's "logical" resolution, not its full resolution. For example, my current screen has a 3840 x 2160 resolution, and I'm using 2x scale mode, so the logical resolution is 1920 x 1080.

When I use `getUserMedia` or `getDisplayMedia` to capture the screen using a `<video>` tag or the `ImageCapture` class, I get a video or image that's 1920 x 1080 and visibly downscaled. Using `ImageCapture` in Chrome, I get a video/image that's the correct resolution.

When I was experimenting with this, I noticed that the video stream metadata (accessed from JS) does have the correct width and height properties, but attaching it to a `<video>` element gives it incorrect `videoWidth` and `videoHeight` values.

### Testcase Gist URL

https://gist.github.com/rf-figma/7bf241daa46134583b82175c6e93b8e9

### Additional Information

Note that I had trouble running this test case inside Fiddle because of OS permissions, so I ended up just running it from the command line.

Also, here's a CodePen you can run in Chrome to see the browser behavior: https://codepen.io/rf-figma/pen/vYoyMRg

**Comment by acemtp** [comment]
I have the same issue and I think everybody on Mac using electron will have the same issue.

I don't know if it's an electron issue or chromium but it's very sad to not be able to have crisp screen capture.

**Comment by acemtp** [comment]
@codebytere why is it not seem as a bug?

**Comment by acemtp** [comment]
It not seems to be related to Electron.

I found some discussion in w3c and chromium about using the logical resolution as source but it's pretty hard to test because:

on Chrome, we can use getDisplayMedia but we cannot use getUserMedia with chromeMediaSource (Uncaught AbortError: Invalid state).

on Electron, we can use getUserMedia but we cannot use getDisplayMedia (Uncaught NotSupportedError: Not supported)

So in the end, I imagine that getDisplayMedia return a physical resolution and getUserMedia return a logical resolution.

@rf-figma did you find a solution or workaround?

**Comment by rf-figma** [comment]
No, I didn't.

**Comment by rf-figma** [comment]
BTW you can use `getDisplayMedia` on Electron, you just have to use `setDisplayMediaRequestHandler` to enable the permission. This behavior with the resolution is the same with either API on Electron.

**Comment by acemtp** [comment]
Indeed, I saw that and tried with getDisplayMedia in electron and it failed too.

I don't see any code in electron that could explain that, so it should be in Chromium but no reason why it's not the same in Chrome.
